
# HG changeset patch
# User John Lindgren <john.lindgren@tds.net
# Date 1264711608 18000
# Node ID bcf2c84660d8674caf7a1dc843b802b2878abae9
# Parent  439b7f5f72685527fd4db9facf369513fed9de16
madplug: Ignore empty string in stream metadata, as noted on AUD-136.

diff -r 439b7f5f7268 -r bcf2c84660d8 src/madplug/input.c
--- a/src/madplug/input.c	Wed Jan 27 23:45:19 2010 -0500
+++ b/src/madplug/input.c	Thu Jan 28 15:46:48 2010 -0500
@@ -396,7 +396,7 @@
 static gchar * get_stream_metadata (VFSFile * file, const gchar * name)
 {
     gchar * raw = aud_vfs_get_metadata (file, name);
-    gchar * converted = (raw != NULL) ? aud_str_to_utf8 (raw) : NULL;
+    gchar * converted = (raw != NULL && raw[0]) ? aud_str_to_utf8 (raw) : NULL;
 
     g_free (raw);
     return converted;

